tests/fgcmcalTestBase.py

```diff
@@ -112,31 +110,26 @@ def _runPipeline(self, repo, pipelineFile, queryString=None,
         ------
         RuntimeError : Raised if the "pipetask" call fails.
         """
-        pipelineArgs = ["run",
-                        "-b", repo,
-                        "-p", pipelineFile]
-
-        if queryString is not None:
-            pipelineArgs.extend(["-d", queryString])
-        if inputCollections is not None:
-            pipelineArgs.extend(["-i", inputCollections])
-        if outputCollection is not None:
-            pipelineArgs.extend(["-o", outputCollection])
-        if configFiles is not None:
-            for configFile in configFiles:
-                pipelineArgs.extend(["-C", configFile])
-        if configOptions is not None:
-            for configOption in configOptions:
-                pipelineArgs.extend(["-c", configOption])
-        if registerDatasetTypes:
-            pipelineArgs.extend(["--register-dataset-types"])
-
-        # CliRunner is an unsafe workaround for DM-26239
-        runner = click.testing.CliRunner()
-        results = runner.invoke(lsst.ctrl.mpexec.cli.pipetask.cli, pipelineArgs)
-        if results.exception:
-            raise RuntimeError("Pipeline %s failed." % (pipelineFile)) from results.exception
-        return results.exit_code
+        butler = SimplePipelineExecutor.prep_butler(repo,
+                                                    inputs=inputCollections,
+                                                    output=outputCollection)
+
+        pipeline = Pipeline.fromFile(pipelineFile)
+        for configFile in configFiles:
+            taskName, fileName = configFile.split(':')
+            pipeline.addConfigFile(taskName, fileName)
+        for configOption in configOptions:
+            taskName, others = configOption.split(':')
+            option, value = others.split('=')
```

Comment:
I'm not keen on the string manipulation here: I feel like these could be dictionaries instead, so you don't have to muck with strings. I found that to be an improvement in jointcal: just pass a dict of the config field:value you want to apply. But you're also specifying different tasks, so maybe it's not as nice? Would a nested dict be bad?